### PR TITLE
op-chain-ops: update script to submit pending withdrawals

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Replaces #4230

Creates a script to send migrated pending withdrawals. It requires the witness scripts generated by `l2geth`.
I didn't find any withdrawals that were broken during the migration.

Closes ENG-2740
Closes ENG-3151

### CODE REVIEW

File: op-chain-ops/cmd/withdrawals/main.go
Comment (protolambda) [inline op-chain-ops/cmd/withdrawals/main.go:641]: do `callFrame.Value` and `wdValue.Text(16)` pad with any zeroes, or are they just hex encoded ints? Need to be sure unexpected padding cannot throw off the comparison.

**Comment by tynes** [comment]
Thanks for the review @protolambda, will push fixes

File: op-chain-ops/cmd/withdrawals/main.go
Comment (trianglesphere) [inline op-chain-ops/cmd/withdrawals/main.go:314]: There's lots of config / parameter checking very far down int the function.

File: op-chain-ops/cmd/withdrawals/main.go
Comment (trianglesphere) [inline op-chain-ops/cmd/withdrawals/main.go:331]: This is more script like, but even for a script, this is a very long function.

File: op-chain-ops/cmd/withdrawals/main.go
Comment (trianglesphere) [reply op-chain-ops/cmd/withdrawals/main.go:492]: IMO this should probably be a trace rather than debug if there are lots of withdrawals here.

File: op-chain-ops/cmd/withdrawals/main.go
Comment (trianglesphere) [inline op-chain-ops/cmd/withdrawals/main.go:518]: This switch / these case statements are good candidates to pull out into their own function.

File: op-chain-ops/cmd/withdrawals/main.go
Comment (tynes) [reply op-chain-ops/cmd/withdrawals/main.go:271]: You are right and this was bothering me so i refactored it a bunch

File: op-chain-ops/cmd/withdrawals/main.go
Comment (tynes) [reply op-chain-ops/cmd/withdrawals/main.go]: you are right that they do not all need to be the same here, its not useful to learn what the problem was

**Comment by tynes** [comment]
Since this changes some types in structs involved in the migration, we should not merge this one until after the migration is complete

File: op-chain-ops/cmd/withdrawals/main.go
Comment (trianglesphere) [inline op-chain-ops/cmd/withdrawals/main.go:212]: What do you mean by "This is just wrong"? Is this a post bedrock withdrawal (the `new`)?

**Review by trianglesphere — COMMENTED** [review]
for my understanding, this script is ran post-bedrock migration to migrate all pending withdrawals? I think it looks good, just one question on a comment.

**Comment by tynes** [comment]
> for my understanding, this script is ran post-bedrock migration to migrate all pending withdrawals? I think it looks good, just one question on a comment.

This script is to be ran against a shadow fork so that we can test finalizing the migrated withdrawals. It asserts that the migrated withdrawals do what we expect

File: op-chain-ops/cmd/withdrawals/main.go
Comment (tynes) [reply op-chain-ops/cmd/withdrawals/main.go:212]: ah sorry, that is a dead comment. good catch

**Comment by tynes** [comment]
@trianglesphere I deleted that dead comment

**Review by trianglesphere — APPROVED** [review]
lgtm

**Pull request merged**
